ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -63,21 +63,21 @@ public ForkChoice(
   }
 
   private void initializeProtoArrayForkChoice() {
-    processHead();
+    processHead().join();
   }
 
-  public void processHead() {
-    processHead(Optional.empty());
+  public SafeFuture<Void> processHead() {
+    return processHead(Optional.empty());
   }
 
-  public void processHead(UInt64 nodeSlot) {
-    processHead(Optional.of(nodeSlot));
+  public SafeFuture<Void> processHead(UInt64 nodeSlot) {
+    return processHead(Optional.of(nodeSlot));
   }
 
-  private void processHead(Optional<UInt64> nodeSlot) {
+  private SafeFuture<Void> processHead(Optional<UInt64> nodeSlot) {
     final Checkpoint retrievedJustifiedCheckpoint =
         recentChainData.getStore().getJustifiedCheckpoint();
-    recentChainData
+    return recentChainData
```

Comment:
Since sometimes we do skip the update ("Skipping head block update..."), should `processHead` return `SafeFuture<Boolean>`?

Reply:
hmm, probably makes sense. There's nowhere that actually uses that info but may as well provide it so long as its simple (ie that we don't then just add a bunch of conversions back to void around the place).